src/redux-reducer-effects.ts

```diff
@@ -37,49 +37,55 @@ const liftReducer = <Msg, S, Task>(reducer: EnhancedReducer<S, Task>, callback:
     }
 }
 
-function createSubject <T>() {
-    type Subscriber = (t: T) => any;
-    const subscribers: Subscriber[] = [];
-    const subscribe = (subscriber: Subscriber) => subscribers.push(subscriber)
-    const onNext = (t: T) => subscribers.forEach(fn => fn(t));
-    const map = <A>(mapper: (t: T) => A) => {
-        const newSubject = createSubject<A>();
-        subscribe(t => newSubject.onNext(mapper(t)))
-        return newSubject;
-    }
-    return { onNext, subscribe, map }
+export interface Observable {
+    subscribe(fn?: any, error?: (err:any) => void, complete?: () => void): any;
+}
+
+export interface Subject<T> {
+    next(t: T): void;
+    flatMap(...xs: any[]): any;
 }
 
-export type TaskRunner<Msg> = <Task>(task: Task) => Promise<Msg>;
+export interface EnhanceOptions {
+    createSubject(): Subject<any>
+    taskRunner: TaskRunner,
+    scheduler?: (fn: () => void) => void;
+};
 
-const enhance = (originalCreateStore: StoreCreator) => {
-    return <S, Task, Msg extends Action>(taskRunner: TaskRunner<Msg>, reducer: EnhancedReducer<S, Task>, initialState: S, enhancer?: StoreEnhancer<S>) => {
+export type TaskRunner = <Task>(task: Task) => Observable;
```

Comment:
We probably want to type check the observable returned here complies with the user's `Action` (or wrapped version, e.g. what was before `Msg`). Should be simple to add that generic to your `Observable` type?